### PR TITLE
Sparse Multivariate Normal Distribution

### PR DESCRIPTION
**Pull request by theo-barfoot**
Implementation of sparse multivariate normal distribution allowing parameterisation of both covariance and precision matrices via LDLt decomposition. 

Currently only implemented rsample function for SparseMultivariateNormal class.

This implementation also required the following changes:
- Adding transpose flag to sparse_triangular_solve, to allow solving A^T X = B
- Implementing _batch_sparse_mv to permit sampling following same approach as pytorch MultivariateNormal

A sparse_eye function was also added to the utils, but was ultimately not needed as a change to the sampling formula for the covariance parameterisation implying a unit diagonal was used.

### CODE REVIEW

**Comment by theo-barfoot** [comment]
torch 1.13 tests are failing, maybe due to the addition "and not ctx.transpose" in the workaround88890 conditions. I'll take a look after some lunch.

File: torchsparsegradutils/sparse_solve.py [hunk 1]
```diff
@@ -103,7 +107,9 @@ def forward(ctx, A, B, upper, unitriangular):
             A = convert_coo_to_csr(A)  # triangular solve doesn't work with sparse coo
             ctx.csr = False
 
-        x = torch.triangular_solve(B.detach(), A.detach(), upper=upper, unitriangular=unitriangular).solution
+        x = torch.triangular_solve(
+            B.detach(), A.detach(), upper=upper, unitriangular=unitriangular, transpose=transpose
+        ).solution
```
Comment (tvercaut) [inline torchsparsegradutils/sparse_solve.py:112]: As discussed, this requires using a workaround88890 approach

**Comment by tvercaut** [comment]
As mentioned, let's list the new features in the main readme:
https://github.com/cai4cai/torchsparsegradutils/blob/main/README.md

**Comment by tvercaut** [comment]
Also, let's bump the version to 0.0.2:
https://github.com/cai4cai/torchsparsegradutils/blob/0f0b2c8f0f5fdb9fb12607f3da0872cffa6dffdc/setup.py#L11

**Pull request merged**